### ISSUE TITLE
Subtitles overlapping 

### ISSUE THREAD

**Issue by Yurovoid** [rebelonion/Dantotsu#334]
While watching episode 2 of demon slayer season 1, there came a certain moment at 11:58 - 11:59 where one character was thinking and other was speaking. So the subtitles of both of them were displayed but they were overlapping. (I used hianime extension)
![Screenshot_2024-04-07-11-08-48-610_ani dantotsu](https://github.com/rebelonion/Dantotsu/assets/144203241/7bbb114a-5c46-4c5f-abc8-08791ec50107)
![Screenshot_2024-04-07-11-08-54-013_ani dantotsu](https://github.com/rebelonion/Dantotsu/assets/144203241/7c853ad5-f37b-4658-a12b-f6dcbffaade3)

**Comment by JohnVictoryz** [comment]
this most likely is a source or a subtitle issue on h!anime side and not an issue with the app have you tried other sources like allanime, yomiroll etc

**Comment by R-iscool** [comment]
Ok sure I'll try other sources and also I checked the same episode on h!anime and it wasn't overlapped.

**Comment by AbandonedCart** [comment]
You changed your subtitle settings, right? What size are you using?

**Comment by R-iscool** [comment]
Default (20)

**Comment by AbandonedCart** [comment]
Did you change the font or anything?

**Comment by R-iscool** [comment]
Yes I did try to change it but its the same.

**Comment by AbandonedCart** [comment]
Alright, then it is a possibility that somewhere in the changes I made for torrents, it ends up fixing it. Let me see if I can backport just the subtitle rewrite and submit that.

**Comment by AbandonedCart** [comment]
I have a feeling the fix is nothing more than
```
subtitles.setApplyEmbeddedStyles(false)
subtitles.setApplyEmbeddedFontSizes(false)
```
which is supposed to be set when you want the custom settings to override the defaults, but the rest won't hurt.

**Comment by R-iscool** [comment]
So this might be implemented in the next update?

**Comment by AbandonedCart** [comment]
Who knows. I submitted the code. The rest is up to them.

**Comment by R-iscool** [comment]
Lastly, could you suggest me some other good sources other than h!anime?

**Comment by JohnVictoryz** [comment]
Idk I use allanime and it works also it looks like yomiroll is not available anymore it was available when I installed dantotsu but not anymore apparently other alternatives can be sflix and animeonsen other than that I can't suggest anything else cause I don't have used anything else I hope it helps

**Comment by AbandonedCart** [comment]
I think the alphas are posted to Discord or you can unzip https://github.com/rebelonion/Dantotsu/actions/runs/8593848048/artifacts/1392735934 and install the compatible one

**Comment by R-iscool** [comment]
I installed alpha and checked it out, and no, it isn't fixed.

**Comment by AbandonedCart** [comment]
I wonder what's different because I'm able to display them normally.

**Comment by R-iscool** [comment]
Is it the same episode and extension (I used hianime and was demon slayer season 1 ep 2 11:58-59)

**Comment by AbandonedCart** [comment]
Yes. The part where the demon is stuck to the tree and is shouting over Tanjiro talking to himself.

**Comment by R-iscool** [comment]
Oh I see. Must be a problem with the version of alpha? I used universal

**Comment by AbandonedCart** [comment]
That should be fine. I'm also running a completely different build. I'm trying to figure out what's in mine that isn't in this one that makes mine work.

**Comment by R-iscool** [comment]
You used hianime right? Also make sure subtitle window isn't on or else it won't look like a previous subtitle is behind another.

**Comment by AbandonedCart** [comment]
Isn't on meaning transparent, right?

**Comment by R-iscool** [comment]
Yes. I'm now starting to think it could be an issue on my device.

**Comment by AbandonedCart** [comment]
You could try going into the backup and restore and doing a backup of everything, wiping the app data, but not selecting player when you do the restore. See if that fixes it. You can always restore the player settings after that, unless that was the cause. Then you may want to set it up manually.

**Comment by R-iscool** [comment]
Can u dm me on discord?
User : r_is_cool19007

**Comment by AbandonedCart** [comment]
I'm not a member of this Discord, sorry.

**Comment by R-iscool** [comment]
Ok, for the last try, could you send me screenshots on how it looks on your end, if possible, so I can check if I have to change somethings.
